### PR TITLE
Update data_extraction.rst

### PR DESCRIPTION
**Pull request by jspeerless**
# Citrine Python PR

## Description 
Updates to table documentation to:
* Clarify that Material UIDs are returned by `IngredientIdentifierInOutput`
* Update the build table examples to include `build_from_config`, `initiate_build`, and `get_by_build_job` methods.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [N/A] I have added tests for 100% coverage
- [N/A] I have written Numpy-style docstrings for every method and class.
- [x] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by jspeerless** [comment]
@maxhutch @latture - Please Review

**Comment by maxhutch** [comment]
Reassigning to @kroenlein for the data team perspective.

File: docs/source/data_extraction.rst
Comment (kroenlein) [inline docs/source/data_extraction.rst:191]: This code runs but issues warnings.  If correcting this is out of scope for this PR, we don't want it to go disappear off our radar.

**Review by kroenlein — CHANGES_REQUESTED** [review]
There's a question of how ambitious the scope on the corrections here should be.

File: docs/source/data_extraction.rst
Comment (jspeerless) [reply docs/source/data_extraction.rst:191]: What's the warning message? I don't remember seeing one, but I often suppress warnings in my workflows.

Comment (jspeerless) [reply docs/source/data_extraction.rst:191]: Just checked, I don't get one using `initiate_build` (using citrine v.0.61 though)

File: docs/source/data_extraction.rst
Comment (jspeerless) [reply docs/source/data_extraction.rst]: I agree with your description. See the latest commit to see the changed language.
I only changed it in two places though (`IngredientIdentifierInOutput` and `IngredientQuantityInOutput`). What's the third you mentioned?

File: docs/source/data_extraction.rst
Comment (jspeerless) [reply docs/source/data_extraction.rst:191]: Still a little confused on this - if it's a deprecated method, we should remove it from the docs. If not, we should make a card to address it. But I can't reproduce the warning you're getting.

Comment (kroenlein) [reply docs/source/data_extraction.rst:191]:
```
<ipython-input-11-6c84f4e184d2>:6: DeprecatedWarning: get_job_status is deprecated. Table build job status does not have a stable format. Please use TableCollection.build_from_config or TableCollection.get_by_build_job
  status = project.table_configs.get_job_status(job.job_id)
```
Issued under both Citrine 0.66 and 0.70.

File: setup.py
Comment (kroenlein) [inline setup.py:60]:
```suggestion
      version='0.72.1',
```

File: docs/source/data_extraction.rst
Comment (kroenlein) [inline docs/source/data_extraction.rst:244]: Lookin' good.

File: docs/source/data_extraction.rst
Comment (jspeerless) [reply docs/source/data_extraction.rst:191]: Oh so it's the part where we query for the job status. Yeah, if you know the correct way to do that that would be great!

File: docs/source/data_extraction.rst
Comment (kroenlein) [inline docs/source/data_extraction.rst:207]: Obsolete reference
```suggestion
You can also use the :class:`~citrine.resources.job.JobStatusResponse` to return the :class:`~citrine.resources.gemtables.GemTable` resource directly with the ``get_by_build_job`` method.
```

File: docs/source/data_extraction.rst
Comment (kroenlein) [inline docs/source/data_extraction.rst:201]:
```suggestion
The return type of the ``initiate_build`` method is a :class:`~citrine.resources.job.JobSubmissionResponse` that contains a unique identifier for the submitted job.
```

**Review by kroenlein — CHANGES_REQUESTED** [review]
Good progress.  Thanks for this cleanup.

File: setup.py
Comment (latture) [reply setup.py:60]: @kroenlein Why make this a minor version bump instead of incrementing the revision number? Also, did you mean to suggest 0.72.0?

Comment (latture) [reply setup.py:60]: Never mind. I see version 0.72.0 is the latest in master.

File: docs/source/data_extraction.rst
Comment (latture) [reply docs/source/data_extraction.rst:191]: @kroenlein Did you look into this? I think the example should be something like:

```python
# 1. Submit the GEM Table build job
job = project.tables.initiate_build(table_config)
# 2. Get the table resource as an object
table = project.tables.get_by_build_job(job)
# 3. Download the table
project.tables.read(table, "./my_table.csv")
```

`get_by_build_job` already waits for the job to complete (and checks the status afterwards), so I don't think we don't need explicit polling.

Comment (jspeerless) [reply docs/source/data_extraction.rst]: @latture I think it's smart to remove the polling, since the other method waits for completion. In this case, I think it's good to separate the `initiate_build` and `get_by_build_job` methods, since running them back to back will result in an error if you don't wait. 

See the a2d55e8 commit, Line 174-198 for my edits.

It still includes a reference to `get_job_status` in the text. Is there another way to tell if the build job has completed or not?

Comment (latture) [reply docs/source/data_extraction.rst]: > since running them back to back will result in an error if you don't wait.

Maybe I'm misunderstanding your comment, but why would you get an error if you run them back to back?

> It still includes a reference to get_job_status in the text. Is there another way to tell if the build job has completed or not?

I don't see a dedicated (non-deprecated) method to get the status. @kroenlein Do you know if one? Given a session and job ID you can get it using

```python
path = 'projects/{}/execution/job-status'.format(project_id)
params = {'job_id': job_id}
response = session.get_resource(path=path, params=params)
status: JobStatusResponse = JobStatusResponse.build(response)
```

but we generally try to avoid explicit session calls.

Comment (kroenlein) [reply docs/source/data_extraction.rst]: I didn't get a chance to run with it.  I would argue we should probably have async demo code present as well, but the synchronous call example was also in the documentation.  The presumed path would be through the resources.job module in citrine-python.

File: docs/source/data_extraction.rst
Comment (kroenlein) [inline docs/source/data_extraction.rst:184]:
```suggestion
```
`get_job_status` is deprecated, as is the entire job status interface.  Short of a feature request to actually allow for user polling of the server, this should be removed from the documentation.  In digging up what would need to get done, at this point we'd either need to craft new methods or have the user roll their own URLs.

**Review by kroenlein — CHANGES_REQUESTED** [review]
Some minor issues; we should be able to put this to bed quickly.

**Review by kroenlein — APPROVED** [review]

**Comment by jspeerless** [comment]
@kroenlein @latture In case you're waiting for me to merge, I am not authorized to do so.

**Comment by kroenlein** [comment]
> @kroenlein @latture In case you're waiting for me to merge, I am not authorized to do so.

@jspeerless Can you bump the version to 0.72.3? I'll merge once that's done.

**Comment by jspeerless** [comment]
@kroenlein done!

**Review by kroenlein — APPROVED** [review]

**Pull request merged**
